src/compiler/emitter.ts

```diff
@@ -1693,8 +1693,20 @@ module ts {
             }
 
             function isExistingName(location: Node, name: string) {
-                return !resolver.isUnknownIdentifier(location, name) ||
-                    (currentScopeNames && hasProperty(currentScopeNames, name));
+                if (!resolver.isUnknownIdentifier(location, name)) {
+                    return true;
+                }
+                if (currentScopeNames && hasProperty(currentScopeNames, name)) {

```

Comment:
Add a newline before the last `if` and the next `var`. Then leave a comment like 

```TypeScript
// Try checking the current scope and all previous scopes to fully
// check whether the a variable with the given name has been declared.
```

Then add to the comment on why the resolver doesn't already cover these cases. From what I can gather, this is for cases where the resolver is unaware of another variable which you have generated and are emitting.